packages/sanity/src/core/store/_legacy/grants/grantsStore.ts

```diff
@@ -65,11 +61,11 @@ async function matchesFilter(
 /** @internal */
 export interface GrantsStoreOptions {
   client: SanityClient
-  currentUser: CurrentUser | null
+  userId: string | null
 }
 
 /** @internal */
-export function createGrantsStore({client, currentUser}: GrantsStoreOptions): GrantsStore {
+export function createGrantsStore({client, userId}: GrantsStoreOptions): GrantsStore {
```

Comment:
I know these are marked `@internal` but that doesn't mean a customer has not taken a dependency on them. I like this approach as it is cleaner, but we may need to be safer. A couple of alternatives could be:
* accept either userId or user.id if passed in
* use reflection to see if `user` is a string and use it as an id, otherwise use it as a User object
* have two separate methods, one that takes an id and one that takes a User object

Am I overthinking this @rexxars ?

Reply:
Also, is there an interface that can be leveraged for the different classes of User?

Reply:
That feels safer. Can I ask, do we need two separate interfaces?